android/app/webviewAssets.gradle

```diff
@@ -7,6 +10,45 @@
  * more information.
  */
 
+/**
+ * Windows support functions.
+ *
+ * On Windows, we have three different possible setups:
+ *   1. Windows Gradle + MSys Git's bash.
+ *   2. Windows Gradle + WSL's bash.
+ *   3. WSL Gradle + WSL's bash.
+ *
+ * Case 3, fortunately, should behave exactly like Linux.
+ *
+ * In cases 1 and 2, however, all paths Gradle provides are backslash-separated.
+ * This is problematic on its own, as the bash script we're invoking doesn't
+ * grok backslashes, so we normalize those on Windows. That's enough for case 1.
+ *
+ * Case 2 is worse, though. MSys's `bash` groks both "C:/" and "C:\"; more
+ * generally, it operates on the Windows filesystem with Windows path names. The
+ * WSL `bash` doesn't -- it's a standard Linux `bash` (or close enough), and it
+ * only sees the Windows disks as ordinary mounts under the lone volume '/'.
+ * Either "C:/" or "C:\" will confuse it terribly.
+ *
+ * And, of course, Gradle has no good way of distinguishing between cases 1 and
+ * 2. As far as it's concerned, `bash.exe` is `bash.exe`.
+ *
+ * Since we can't predict the environment of our callee, we can't use absolute
+ * paths. Fortunately, with only a little extra work in the bash script, we
+ * _can_ use relative paths, letting the script derelativize them appropriately.
+ */
+def normalizePath(String path) {
+    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+        return path.replace('\\', '/')
```

Comment:
Hmm... is there a possible case where there actually is already a `/` in one of the path components? :grin: If so, that will surely cause confusion down the line.

Not sure there'd be anything sensible to do with that other than throw an error.

Reply:
Nope! NTFS and FAT32 both disallow `0x2F` as a character in file and directory names.

Reply:
Great! Good to know, thanks.

A one-line comment to that effect might be nice in case someone has the same question (though I'm sure many people already know it well.)